cloudwatch_metrics_collector/src/metrics_collector.cpp

```diff
@@ -53,7 +60,7 @@ void MetricsCollector::Initialize(std::string metric_namespace,
                          const Aws::Client::ClientConfiguration & config,
                          const Aws::SDKOptions & sdk_options,
                          const Aws::CloudWatchMetrics::CloudWatchOptions & cloudwatch_options,
-                         const std::vector<std::string> & topics,
```

Comment:
I thought we wanted to be fully backwards compatible. Instead of changing the existing function's signature I would overload it. The version with TopicInfo would contain the implementation, while the old one would simply construct a TopicInfo vector and pass through to the new function.

Reply:
Is the `MetricsCollector` class one in particular that we need to maintain backwards compatibility for? My impression was that `MetricsCollector` is not a public API. I would not add more APIs than absolutely necessary.

Reply:
Is anyone currently using this class directly? *probably* not. But the function is still defined as a public interface, and maintaining backwards compatibility costs us just a few lines of trivial code.

Reply:
Yes, it is trivial and is a few lines, but it can make the API ugly pretty quickly.

Anyway, I'll add the backwards compatibility API this time.